example/exampleContour.html

```diff
@@ -68,56 +68,60 @@
       .mapboxToken(mapboxToken) // need a mapbox accessToken for loading the tiles
       .popupSearchRadius(2)
       .useGeoTypes(true) // need for projecting geo column using "mercator_map_projection"
-
+      
+      const intervalTypeFieldName = "is_major"
+      
       const contourPolygonLayer = dc.rasterLayer("polys")
       .crossfilter(crossFilter)
       .setState({
-          data: [
-            {
-              type: "contour",
-              table: config.table,
-              source: config.table,
-              contour_value_field: config.value,
-              lat_field: config.lat_field,
-              lon_field: config.lon_field,
-              agg_type: 'AVG',
-              fill_agg_type: 'AVG',
-              bin_dim_meters: config.bin_dim_meters,
-              neighborhood_fill_radius: 0,
-              fill_only_nulls: false,
-              flip_latitude: false,
-              minor_contour_interval: config.minor_contour_interval,
-              major_contour_interval: config.major_contour_interval,
-              contour_offset: 0,
-            }
-          ],
-          transform: {
-            projection: "mercator_map_projection",
-          },
-          mark: {
-            type: "poly",
-            strokeColor: "white",
-            strokeWidth: 0,
-            fillColor: "",
-            lineJoin: "miter",
-            miterLimit: 10
-          },
-          encoding: {
-            geocol: "contour_polygons",
-            color: {
-              type: "linear",
-              domain: [0, 6000],
-              range: ["#FF0000", "#0000FF"],
-              opacity: 0.50
-            }
-          },
-          enableHitTesting: false
-        })
+        data: [
+        {
+          type: "contour",
+          table: config.table,
+          source: config.table,
+          contour_value_field: config.value,
+          lat_field: config.lat_field,
+          lon_field: config.lon_field,
+          agg_type: 'AVG',
+          fill_agg_type: 'AVG',
+          bin_dim_meters: config.bin_dim_meters,
+          neighborhood_fill_radius: 0,
+          fill_only_nulls: false,
+          flip_latitude: false,
+          contour_offset: 0,
+          interval: {
+            intervalTypeFieldName,
```

Comment:
`intervalTypeFieldName` is an ok name. Problem is that it still doesn't quite describe the output. The output is a boolean (or more simplistically, a bit flag) that determines whether or not it is a major or minor interval. Maybe it should be named "isMajorColumnName" or "isMajorField" or "isMajorTag" ... "intervalType" implies that the domain below should read:

```
  domain: ["major", "minor"]
  ...
```

As I write that down, I now wonder if that's what we should do -- use a string-type identifier, rather than an "is_major" true/false.

We can still do that in the `CASE` as the string will auto-convert to a string dictionary (as long as the table-function output can handle that -- experience thus far is it is a bit flaky yet, but in normal cases it would work just fine).

Something to consider. Otherwise looks good to me.

Reply:
Makes sense about the `is*` naming, I can def do that bit

re: string identifiers, I feel like these identifiers would also have to be specified as part of `data`, or we'd be dealing with more implicit/magic strings ("major"/"minor"), which to me is starting to look a bit more complex than it needs to be/overly configurable.

Reply:
I can do the string identifier way too (assuming it works with tf), just my 2 cents

Reply:
When I write this down:

```
{
  type: "contour",
  intervals: {
    typeFieldName: "interval_type",
    major: <major interval>,
    minor: <minor interval>,
  }
}
...

color: {
  field: "interval_type",
  type: "nominal",
  scale: { 
    domain: ["minor", "major"], 
    range: ["#999999", "#999999"]
  }
}
```

makes some sense to me.

The `major`, `minor` keys in the `intervals` block could be used directly as field names to avoid further verbosity/redundancy, so if you could change those names to:

```
{
  type: "contour",
  intervals: {
    typeFieldName: "interval_type",
    intervalA: <major interval>,
    intervalB: <minor interval>,
  }
}
...

color: {
  field: "interval_type",
  type: "nominal",
  scale: { 
    domain: ["intervalA", "intervalB"], 
    range: ["#999999", "#999999"]
  }
}
```

if you wanted. It does give way to allowing more interval types, if one wants, tho under the limitation that they need to be divisible by a least-common-denominator.

Anyway, sorry, we're snowballing here in a way, but it's important to decide because such things are harder to change later.

Reply:
Ahh using the keys as the values does make more sense, though still not immediately clear (to n00b me anyways) that that is whats happening. I like the flexibility angle though, i'll give this a shot

Reply:
It may make life much easier if we eventually add user defined intervals

Reply:
So actually (realized when implementing this) I think we would need to provide more information than just an interval number to be able to identify > 2 interval types or we'll run into collisions pretty quick, and i'm not sure i want to go down the road of how to resolve those/etc in this PR. 

I could see something in the future for custom intervals like:
{
  intervals: {
    typeFieldName: "interval_type",
    intervalA: [0, 50],
    intervalB: [50, 100]
    ....
  }
}```

Reply:
Or if you had an idea of what the SQL would look like for having > 2 intervals: 
```
  intervals: {
    typeFieldName: "interval_type",
    intervalA: <major interval>,
    intervalB: <minor interval>,
    intervalC: <other interval>,
  }
```

I'm having a hard time figuring out what that would look like

Reply:
@vastcharade I'd like to get this one merged ASAP, but you have requested changes still. If you can give it another pass through today that would be much appreciated! Given the timeline i'm going to defer the additional changes in the comments above until we have a need for them.

Reply:
ok, deferment is a good plan. I'll revisit this now.

But just to clarify on one point, although something like this:

```
{
  intervals: {
    typeFieldName: "interval_type",
    intervalA: [0, 50],
    intervalB: [50, 100]
    ....
  }
}
```

would be ideal, we are stuck by the limitation that the contour function only takes a single contour interval as an argument. Not a set of intervals. So when dealing with multiple intervals, you have to take the smallest of the intervals and make sure the others are divisible by it.

Then you use the least-common-denominator interval for the contour interval argument, and then expand the `CASE` statement outside the table function to include the multiple variants, so if you had 4 defined intervals, the case would look like so:

```
CASE mod(cast(contour_values as int), <max interval>) WHEN 0 THEN "intervalA" WHEN 1 THEN "intervalB" WHEN 2 THEN "intervalC" ELSE "intervalD" END as interval_type
```

and then your scales would look like:

```
color: {
  field: "interval_type",
  type: "nominal",
  scale: { 
    domain: ["intervalA", "intervalB", "intervalC", "intervalD"], 
    range: [<color A>, <color B>, <color C>, <color D>]
  }
}
```

In any event, we can defer for now. I will look re-review what you've got.

Reply:
🙏 🙇

Reply:
The big assumption would be that there is a common denominator to all of the interval sizes specified, which _I think_ is out the window when we're talking about custom interval ranges, but with only major/minor we are enforcing in immerse (and charting now)